File: backend/app/models/record.py
from datetime import datetime

from sqlalchemy import DateTime, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Record(Base):
    __tablename__ = "records"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    stream_id: Mapped[str] = mapped_column(String(100), index=True, nullable=False)
    file_path: Mapped[str] = mapped_column(String(500), nullable=False)
    file_name: Mapped[str] = mapped_column(String(255), nullable=False)
    duration_seconds: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    file_size_bytes: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    started_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    ended_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)

File: frontend/src/api/record.ts
import { apiGet } from "./client";

export interface RecordItem {
  id: number;
  stream_id: string;
  file_path: string;
  file_name: string;
  duration_seconds: number;
  file_size_bytes: number;
  started_at?: string | null;
  ended_at?: string | null;
  created_at: string;
}

export function listRecords() {
  return apiGet<RecordItem[]>("/records");
}
